src/main.ts

```diff
@@ -33,9 +33,64 @@ import { environment } from './environments/environment';
 
 import '@pharos/core/web-components/pharos-tooltip';
 
-if (environment.production) {
-  enableProdMode();
+const FLAGS_QUERY = `
+  query AiwFlagList($flagsFlagList: [String]) {
+    flags(flagList: $flagsFlagList) {
+      enabled
+    }
+  }
+`;
+const REDIRECT_FLAG = "artstor_client_redirection";
+const OPTIONS = {
+  operationName: 'AiwFlagList',
+  query: FLAGS_QUERY,
+  variables: {
+    flagsFlagList: [REDIRECT_FLAG],
+  },
+};
+
+const initializeApp = () => {
+  if (environment.production) {
+    enableProdMode();
+  }
+
+  platformBrowserDynamic().bootstrapModule(AppModule)
+    .catch(err => console.log(err));
 }
 
-platformBrowserDynamic().bootstrapModule(AppModule)
-  .catch(err => console.log(err));
+fetch('/ui/data-fetch/gateway', {
+  method: 'POST',
+  headers: {
+    'authorization': 'aiw-ui',
+    'Content-Type': 'application/json',
+  },
+  body: JSON.stringify(OPTIONS),
+})
+  .then(response => response.json())
+  .then(flagData => {
+    const flagStates = flagData.data ? flagData.data : null;
+    const enabledFlags = flagStates ? flagStates.flags.enabled : [];
+    const doRedirect = enabledFlags.includes(REDIRECT_FLAG);
+
+    const currentRequest = window.location.href;
+
+    if (currentRequest.includes('/#/')) {
```

Comment:
I think we might have broken any urls that don't have the `/#/` because we never end up calling `initializeApp()` if this condition fails? I think this is evidenced by the confusion Mike and I ran up against when trying to load `https://stage.artstor.org` vs `https://stage.artstor.org/#/`. 

The only other url I can think of without the hash are the `/public` ones, and I am curious how those ones are going to work now for the next month and a half. I know they are server side rendered, but they might get tripped up now too by this condition when they go to render server side

Could a solution be to just add an `else` condition to this that also calls `initializeApp()`?

Reply:
I think you're right, yep.